include/ncengine/ecs/Transform.h

```diff
@@ -145,6 +145,9 @@ class Transform final : public ComponentBase
         /** @brief Apply a rotation about an axis to local rotation */
         void Rotate(const Vector3& axis, float radians);
 
+        /** @brief Apply a rotation about an axis passing through a point. */
```

Comment:
You might not have figured out exactly the 'how' yet in how the DX Math functions get this to work but - having a Vector3 as an axis doesn't seem possible and makes this method confusing. Can you add info on what is happening under the hood (if you have it?)